R2API/LanguageAPI.cs

```diff
@@ -198,5 +211,221 @@ public static class LanguageAPI {
+        /// Adds multiple temporary language tokens, and corresponding values, to all languages. Language-specific tokens, as well as overlays added later in time, will take precedence. Call LanguageOverlay.Remove() on the result to remove your changes to these language tokens.
+        /// </summary>
+        /// <param name="tokenDictionary">dictionaries of key-value (eg ["mytoken"]="mystring")</param>
+        /// <returns>A LanguageOverlay representing your language addition/override; call .Remove() on it to undo the change.</returns>
+        public static LanguageOverlay AddOverlay(Dictionary<string, string> tokenDictionary) {
+            var overlay = new LanguageOverlay(tokenDictionary.Select(kvp => {return new OverlayTokenData(kvp.Key, kvp.Value);}).ToArray());
+            overlay.Add();
+            return overlay;
+        }
+
+        /// <summary>
+        /// Adds multiple temporary language tokens, and corresponding values, to a specific language. Overlays added later in time will take precedence. Call LanguageOverlay.Remove() on the result to remove your changes to these language tokens.
+        /// </summary>
+        /// <param name="tokenDictionary">dictionaries of key-value (eg ["mytoken"]="mystring")</param>
+        /// <param name="language">Language you want to add this to</param>
+        /// <returns>A LanguageOverlay representing your language addition/override; call .Remove() on it to undo the change.</returns>
+        public static LanguageOverlay AddOverlay(Dictionary<string, string> tokenDictionary, string language) {
+            var overlay = new LanguageOverlay(tokenDictionary.Select(kvp => {return new OverlayTokenData(kvp.Key, kvp.Value, language);}).ToArray());
+            overlay.Add();
+            return overlay;
+        }
+        
+        /// <summary>
+        /// Adds multiple temporary language tokens, and corresponding values, to mixed languages. Overlays added later in time will take precedence. Call LanguageOverlay.Remove() on the result to remove your changes to these language tokens.
+        /// </summary>
+        /// <param name="languageDictionary">dictionary of languages containing dictionaries of key-value (eg ["en"]["mytoken"]="mystring")</param>
+        /// <returns>A LanguageOverlay representing your language addition/override; call .Remove() on it to undo the change.</returns>
+        public static LanguageOverlay AddOverlay(Dictionary<string, Dictionary<string, string>> languageDictionary) {
+            var overlay = new LanguageOverlay(
+                languageDictionary.SelectMany(subdict => {
+                    return subdict.Value.Select(kvp => {
+                        return new OverlayTokenData(kvp.Key, kvp.Value, subdict.Key);
+                    });
+                }).ToArray());
+
+            overlay.Add();
+            return overlay;
+        }
+        
+        private static OverlayTokenData[] LoadOverlayTokensFromFile(string file) {
```

Comment:
Minor point but i prefer to return List<T> rather than T[] when possible if performance is the same (under the hood we're using a list here anyways)